Assets/Scripts/GameManagement/DataPersistence/DataPersistenceManager.cs

```diff
@@ -0,0 +1,96 @@
+using Inventory;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using static UnityEngine.InputSystem.InputAction;
+
+public class DataPersistenceManager
+{
+    private static DataPersistenceManager instance = null;
+    private static readonly object instanceLock = new object(); //thread-safe for co-routines
+
+    private InputAction saveAction;
+    private InputAction loadAction; //only needed for testing
+
+    public event Action onSaveTriggered;
+    public event Action onLoadTriggered;
+
+    private FileDataHandler fileDataHandler;
+    private string fileName = "AngelInsolventSaveFile.txt";
+
+    DataPersistenceManager()
+    {
+        this.fileDataHandler = new FileDataHandler(Application.persistentDataPath, fileName);
+        saveAction = new InputAction(binding: "<Keyboard>/f5"); //should make this serializable somewhere for multiple platforms' inputs
+        saveAction.performed += SaveGame;
+        saveAction.Enable();
+
+        loadAction = new InputAction(binding: "<Keyboard>/f6");
+        loadAction.performed += LoadGame;
```

Comment:
These actions never get unbound, might consider adding a destructor that unbinds them just to be safe: https://learn.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2012/66x5fx1b(v=vs.110)?redirectedfrom=MSDN